primap2/pm2io/_data_reading.py

```diff
@@ -1434,8 +1437,15 @@ def preferred_unit(entity: str, units: List[str], gwp_to_use: Optional[str]) ->
             ureg[native_unit], *conversion_contexts
         ):
             return native_unit
+        else:
+            # try to convert to Gg of the substance present in the unit
+            # extract the substance from the fallback unit
+            substance = unit_fallback.split(" ")[1]
```

Comment:
This is not robust. There is no need that the input will be of the form "<mass> <substance> / <time>", it could be also of the form "<substance> <mass> / <time>" or any other number of things. I think if you really need to go down this rabbit hole, you should just assume `gwp_unit = "Gg CO2 / yr"` and try that (note that please don't re-use the `native_unit` variable here, because it definitely is *not* the native unit, the native unit is in this context the mass unit, not the gwp unit (see above, where `native_unit = "Gg " + entity + " / yr"`).

Reply:
True. That probably comes from thinking only about my use cases where I use a defined format.